server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/RemoveRegexMappingRoutes.java

```diff
@@ -0,0 +1,82 @@
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.routes;
+
+import static spark.Spark.halt;
+
+import javax.inject.Inject;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+
+import org.apache.james.rrt.api.RecipientRewriteTable;
+import org.apache.james.rrt.api.RecipientRewriteTableException;
+import org.apache.james.rrt.lib.MappingSource;
+import org.apache.james.webadmin.Constants;
+import org.apache.james.webadmin.Routes;
+import org.apache.james.webadmin.utils.ErrorResponder;
+import org.eclipse.jetty.http.HttpStatus;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import io.swagger.annotations.Api;
+import spark.HaltException;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+@Api(tags = "Remove Regex")
+@Path(RemoveRegexMappingRoutes.BASE_PATH)
+@Produces(Constants.JSON_CONTENT_TYPE)
+public class RemoveRegexMappingRoutes implements Routes {
+    static final String BASE_PATH = "/mappings/regex";
+    private final RecipientRewriteTable recipientRewriteTable;
+
+    @Inject
+    @VisibleForTesting
+    RemoveRegexMappingRoutes(RecipientRewriteTable recipientRewriteTable) {
+        this.recipientRewriteTable = recipientRewriteTable;
+    }
+
+    @Override
+    public String getBasePath() {
+        return BASE_PATH;
+    }
+
+    @Override
+    public void define(Service service) {
+        service.delete(BASE_PATH + "/" + ":mappingSource" + "/targets/" + ":regex",
+            this::removeRegexMapping);
+    }
+
+    public HaltException removeRegexMapping(Request request, Response response) {
+        try {
+            MappingSource mappingSource = MappingSource.parse(request.params("mappingSource"));
+            String regex = request.params("regex");
+            recipientRewriteTable.removeRegexMapping(mappingSource, regex);
+        } catch (RecipientRewriteTableException e) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
```

Comment:
it should response bad request when invalid regex rather than internal error. I think you should follow what the Add regex mapping route has done by:
 - create new specific exception for invalid regex in RecipientRewriteTable
 - throw it when storing invalid regex in RecipientRewriteTable
 - catch that exception at the route layer, then response 400